cfep-05.md

```diff
@@ -0,0 +1,135 @@
+anaconda.org so that the more adventurous users can try out these pre-release
+packages.
+
+## Motivation
+
+Pypi allows package maintainers to upload "prerelease" builds which users can
+then install via
+
+```
+pip install PACKAGE_NAME --pre
+```
+
+The same semantics are slightly more cumbersome within the `conda` ecosystem but
+are still manageable.  The analogous command with `conda` is (for example):
+
+```
+conda install -c conda-forge/label/rc_matplotlib matplotlib
+```
+
+We, as the conda-forge community need to arrive at a consensus for what labels
+we will affix to packages so that there is a consistent user experience.
+
+## Backwards Compatibility
+
+There are only minimal concerns of backwards compatibility.  There are a few
+relevant labels that package maintainers have affixed to their pre-release
+packages. These are "pre" (1), "prerelease" (2), "rc" (2) and "test" (1).  The
+numbers in parentheses represent the number of projects which have released
+these pre-release packages.  With six total projects using non-main labels, I do
+not think we have a serious concern with backwards compatibility due to changes
+that come about via this proposal.
+
+## Specification
+
+Mark the pre-release package on anaconda.org as "dev" or "rc". The semantics of
+these labels should generally follow the
+[guidelines](https://docs.python.org/devguide/devcycle.html#stages) that Python
+itself follows.
+
+- "rc": [Beta](https://docs.python.org/devguide/devcycle.html#beta) and [Release
+  Candidate](https://docs.python.org/devguide/devcycle.html#release-candidate-rc)
+  (RC). No new features. Bugfix only.
+
+- "dev": [Pre-Alpha](https://docs.python.org/devguide/devcycle.html#pre-alpha)
+  and [Alpha](https://docs.python.org/devguide/devcycle.html#alpha). These are
+  still packages that could see substantial changes
+  between the dev version and the final release.
+
+If you wish to add numbers to your "dev" or "rc" you should follow the
+[guidelines](http://conda.pydata.org/docs/spec.html#build-version-spec) put
+forth by Continuum regarding version sorting in `conda`. Also see the [source
+code for conda
+4.2.13](https://github.com/conda/conda/blob/4.2.13/conda/version.py#L93-L119).
+The tl;dr here is that conda sorts as follows:
+
+```
+< 1.0
+< 1.1dev1    # special case 'dev'
+< 1.1.0dev1  # special case 'dev'
+== 1.1.dev1   # 0 is inserted before string
+< 1.1.0rc1
+< 1.1.0
+```
+
+So make sure that you ***tag*** your package in such a way that the package name
+that conda-build spits out will sort the package uploaded with an "rc" label
+higher than the package uploaded with the "dev" label.
+
+To create a `dev` or `rc` package a PR can be issued into the `dev` or `rc` branch of the
+feedstock.
+This branch must change the `recipe/conda_build_config.yaml` to point to the `dev_<package_name>` or `rc_<package_name>`.
+
+For example, matplotlib rc releases would include:
+```yaml
+...
+channel_targets:
+  - conda-forge rc_matplotlib
```

Comment:
Can you also add an example for how to use a dev package in another dev package?

Reply:
Is this something we want to support? Would users get the correct env if they don't have both dev channels in their .condarc/cli call?

Reply:
They won't, but the user should be given specific instructions either way. I think we should support it or say that we don't.

Reply:
I would lean toward not supporting this for now, seeing if people need it and then revisiting. This way we are more likely to provide a stable system day one.

Reply:
I'd need it and I saw some people asking about it in gitter, but can't find it right now.

Reply:
Would you like to push up the language you want to see?

Reply:
We also need to make sure that these extra channels don't leak into downstream `main` builds. Maybe add a poison pill to the build system (unless the linter can see if you PRed into master) that doesn't build on master if you pull from `rc/dev`.

Reply:
Shall I add a section like `Work Items`?

Reply:
That would be great! Thank you!

Reply:
Have a look and feel free to change it.